README.md
# Filament Bookmarks for Filament Admin

[![Latest Version on Packagist](https://img.shields.io/packagist/v/stafe-group-ab/filament-bookmarks-menu.svg?style=flat-square)](https://packagist.org/packages/stafe-group-ab/filament-bookmarks-menu)
[![Total Downloads](https://img.shields.io/packagist/dt/stafe-group-ab/filament-bookmarks-menu.svg?style=flat-square)](https://packagist.org/packages/stafe-group-ab/filament-bookmarks-menu)

![Screen shot of Filament Bookmarks Menu](./images/bookmarks-menu.png)

![Screen shot of Filament User Menu](./images/add-bookmark-menu.png)

This Filament Plugin will add a bookmarks menu to your Filament Admin application. You can setup global items in this menu or let users add their own items.
The plugin will render an add & remove bookmark icon on top of all pages where you want it.

## Installation

You can install the package via composer:

```bash
composer require stafe-group-ab/filament-bookmarks-menu
```

You can publish and run the migrations with:

```bash
php artisan vendor:publish --tag="filament-bookmarks-menu-migrations"
php artisan migrate
```

You can publish the config file with:

```bash
php artisan vendor:publish --tag="filament-bookmarks-menu-config"
```

This is the contents of the published config file, here you can alter icons, colors and more with ease.

```php
return [
    'bookmark_icon' => 'heroicon-o-bookmark',
    'bookmark_class' => 'w-5 h-5 cursor-pointer text-gray-700 dark:text-gray-200',
    'add_bookmark_icon' => 'heroicon-o-bookmark',
    'remove_bookmark_icon' => 'heroicon-s-bookmark',
    'add_bookmark_class' => 'w-6 h-6 cursor-pointer text-danger-700 dark:text-gray-200',
    'remove_bookmark_class' => 'w-6 h-6 cursor-pointer text-danger-700 dark:text-gray-200',
    'add_bookmarks_by_users' => true,
    'exclude_pages' => ['admin', 'bookmarks-menus'],
    'notification_add_icon' => 'heroicon-o-bookmark',
    'notification_remove_icon' => 'heroicon-o-bookmark',
    'notification_add_color' => 'success',
    'notification_remove_color' => 'danger',
    'show_resource_in_navigation' => false,
    'delete_icon' => 'heroicon-o-trash',
    'delete_class' => 'w-5 h-5 cursor-pointer text-danger-700 dark:text-gray-200',
    'delete_text_class' => 'cursor-pointer text-white bg-black dark:text-gray-200',
    'show_private_label_in_menu' => false,
    'show_global_label_in_menu' => false,
];
```

Optionally, you can publish the views using

```bash
php artisan vendor:publish --tag="filament-bookmarks-menu-views"
```

Optionally, you can publish the translation files, we will ship in English only

```bash
php artisan vendor:publish --tag="filament-bookmarks-menu-translations"
```

## Usage
To offer your logged in users the ability to add favorites to their personal bookmarks menu
make sure you go to the config file and set the below. That will make the plugin
render the add / remove bookmark icon feature on all pages.
```php
 'add_bookmarks_by_users' => true,
```
### Delete items in the menu
Now your users will see a svg trash icon next to all their private menu items. You can config this icon
and colors in the config file. You can also set the label in the updated lang files.
```php
    'delete_icon' => 'heroicon-o-trash',
    'delete_class' => 'w-5 h-5 cursor-pointer text-danger-700 dark:text-gray-200',
    'delete_text_class' => 'cursor-pointer text-white bg-black dark:text-gray-200',
```
### Displaying labels in the bookmarks menu?
Now you can enable / disable the Global & Private labels in the Bookmarks Menu in the config file.
```php
   'show_private_label_in_menu' => false,
    'show_global_label_in_menu' => false,
```
### Excluding pages
In almost all cases there will be pages that you want to exclude from
this plugin. There is an array in the config file where you can add
the exclusions you want. It will exclude in the ending string in all
urls in your Filament Admin. So if you exclude edit below no edit pages 
will be available in the plugin.
```php
'exclude_pages' => ['admin', 'bookmarks-menus'],
```


## Testing

```bash
composer test
```

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information on what has changed recently.

## Contributing

Please see [CONTRIBUTING](CONTRIBUTING.md) for details.

## Security Vulnerabilities

Please review [our security policy](../../security/policy) on how to report security vulnerabilities.

## Credits

- [Andreas Kviby, Stafe Group](https://github.com/STAFE-GROUP-AB)
- [All Contributors](../../contributors)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `filament-bookmarks-menu` will be documented in this file.

## Spanish package and user model - 2023-06-03

Thanks for your help with this package.

## Adding Laravel 10 compatibility  - 2023-02-26

Adding Laravel 10 compatibility

## Italian language files  - 2023-02-07

Thanks a lot for the Italian language files

## Model fixed in component - 2023-01-09

Model was not found when deleting a bookmark.

## New features - 2023-01-04

- You can now configure if you want to show the labels Global / Private in the menu for your users.
- Your users will se a trash in the menu to the right, if they click it they must confirm the deletion if the selected menu item.
- Readme updated with instructions on how to use these features.

## Resource can be hidden - 2023-01-01

Due to user wish you can now setup in the config file if you want the resource to be visible in the navigation tree.

## Bugfixes - 2022-12-20

The global search attribute was label and it should be menu_label. Thanks for the report Thijmen#8478 on Discord.

## Updated language files and UI on Dropdown - 2022-12-11

We added Swedish to language files.
We added filament dropdown header to UI of the dropdown menu.


config/filament-bookmarks-menu.php
<?php

// config for STAFEGROUPAB/FilamentBookmarksMenu
return [

    // Icons to use for the icon you might add to your pages in
    // Filament Admin. Also the colors used for active and inactive.
    'bookmark_icon' => 'heroicon-o-bookmark',
    'bookmark_class' => 'w-5 h-5 cursor-pointer text-gray-700 dark:text-gray-200',
    'add_bookmark_icon' => 'heroicon-o-bookmark',
    'remove_bookmark_icon' => 'heroicon-s-bookmark',
    'add_bookmark_class' => 'w-6 h-6 cursor-pointer text-danger-700 dark:text-gray-200',
    'remove_bookmark_class' => 'w-6 h-6 cursor-pointer text-danger-700 dark:text-gray-200',
    'add_bookmarks_by_users' => true,
    'exclude_pages' => ['admin', 'bookmarks-menus'],
    'notification_add_icon' => 'heroicon-o-bookmark',
    'notification_remove_icon' => 'heroicon-o-bookmark',
    'notification_add_color' => 'success',
    'notification_remove_color' => 'danger',
    'show_resource_in_navigation' => true,
    'delete_icon' => 'heroicon-o-trash',
    'delete_class' => 'w-5 h-5 cursor-pointer text-danger-700 dark:text-gray-200',
    'delete_text_class' => 'cursor-pointer text-white bg-black dark:text-gray-200',
    'show_private_label_in_menu' => false,
    'show_global_label_in_menu' => false,
    'user_model' => env('USER_MODEL',App\Models\User::class)
];
